ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlTarget.java

```diff
@@ -255,6 +261,18 @@ private void executeRequestAsync(
     return KsqlClientUtil.toRestResponse(response, path, mapper);
   }
 
+  private <T> CompletableFuture<RestResponse<T>> executeAsync(
+      final HttpMethod httpMethod,
+      final String path,
+      final Object requestBody,
+      final Function<ResponseWithBody, T> mapper,
+      final BiConsumer<HttpClientResponse, CompletableFuture<ResponseWithBody>> responseHandler
+  ) {
+    final CompletableFuture<ResponseWithBody> vcf =
+        execute(httpMethod, path, requestBody, responseHandler);
+    return vcf.thenApply(response -> KsqlClientUtil.toRestResponse(response, path, mapper));
```

Comment:
Is `thenApply` asynchronous? We don't want to block for the response. In fact, `Heartbeat` agent doesn't even care about the response.

Reply:
Discussed offline, as long as we don't call `get` on the future , all other callbacks are non-blocking so we are good